lib/segment/src/index/field_index/numeric_index/mod.rs

```diff
@@ -325,6 +340,43 @@ impl<T: Encodable + Numericable + Default> NumericIndexInner<T> {
     pub fn values_is_empty(&self, idx: PointOffsetType) -> bool {
         self.values_count(idx) == 0
     }
+
+    pub fn values_by_key(&self, key: &T) -> Box<dyn Iterator<Item = PointOffsetType> + '_> {
```

Comment:
Naming might be confusing.

We iterate points by payload values, so why don't we call it `point_ids_by_value` instead?